### PR TITLE
Add coding conventions

### PR DESCRIPTION
**Pull request by kylepw**
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Improvement

---

### Description of your *pull request* and other information
I added some coding conventions based on maintainers' reoccurring comments on pull requests. Issues included: inline values, collapsing fallbacks, and trailing parentheses. My hope is that these additions will save contributor's and maintainer's time.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -366,3 +366,79 @@ duration = float_or_none(video.get('durationMs'), scale=1000)
 view_count = int_or_none(video.get('views'))
 ```
 
+### Inline values
+
+ Always inline values.
```
Comment (dstftw) [inline CONTRIBUTING.md]: This is basically incorrect. Extracting variables is fine for eliminating code duplication and improving readability for complex expressions. It's not fine to extract variables used only once and move them to the opposite part of extractor file making linear reading flow problematic.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -366,3 +366,79 @@ duration = float_or_none(video.get('durationMs'), scale=1000)
...
+
+#### Example
+
+A good rule of thumb is not to create variables that will only be used once.
```
Comment (dstftw) [inline CONTRIBUTING.md]: >only be used once

This should be in definition not in example.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -366,3 +366,79 @@ duration = float_or_none(video.get('durationMs'), scale=1000)
...
+    json_data.get('description'))
+```
+
+Incorrect:
+
+```python
+playlist_vids = json_data.get('videos', [])
+playlist_id = json_data.get('id')
+playlist_title = json_data.get('name')
+playlist_description = json_data.get('description')
```
Comment (dstftw) [inline CONTRIBUTING.md]: Examples should be simple and concise. Here you have the same pattern occurring four times that is superfluous.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kylepw) [reply CONTRIBUTING.md]: Okay, thanks. I reworded it based on what you wrote.

File: CONTRIBUTING.md (hunk 2 above)
Comment (kylepw) [reply CONTRIBUTING.md]: Fixed.

File: CONTRIBUTING.md (hunk 3 above)
Comment (kylepw) [reply CONTRIBUTING.md]: Understood. I changed it to use one variable assignment instead of four.

**Comment by kylepw** [comment]
@dstftw Requested changes made. Let me know if this works.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -366,3 +366,74 @@ duration = float_or_none(video.get('durationMs'), scale=1000)
...
+#### Example
+
+Correct:
+
+```python
+return self.playlist_result(
+    [self._parse_brightcove_metadata(vid, vid.get('id'), headers)
+        for vid in json_data.get('videos', []) if vid.get('id')],
+    json_data.get('id'), json_data.get('name'),
+    json_data.get('description'))
```
Comment (dstftw) [inline CONTRIBUTING.md]: As already pointed out, there should be more simple example with minimal unrelated code. Like:

Correct:
```python
title = self._search_regex(
    r'<span[^>]+class="title"[^>]*>([^<]+)', webpage, 'title')
```

Incorrect:
```python
_TITLE_RE = r'<span[^>]+class="title"[^>]*>([^<]+)'
... some lines of code ...
title = self._search_regex(_TITLE_RE, webpage, 'title')
```

Comment (kylepw) [reply CONTRIBUTING.md]: Okay, thanks. I simplified the example.
